src/lib/AudioUltra/Controls/Html5Player.ts

```diff
@@ -0,0 +1,82 @@
+import { WaveformAudio } from '../Media/WaveformAudio';
+import { Player } from './Player';
+
+export class Html5Player extends Player {
+  init(audio: WaveformAudio) {
+    super.init(audio);
+
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.on('resetSource', this.handleResetSource);
+
+    this.audio.el.addEventListener('play', this.handlePlayed);
+    this.audio.el.addEventListener('pause', this.handlePaused);
+  }
+
+  destroy() {
+    super.destroy();
+
+    if (this.audio?.el) {
+      this.audio.el.removeEventListener('play', this.handlePlayed);
+      this.audio.el.removeEventListener('pause', this.handlePaused);
+    }
+  }
+
+  protected playAudio(start?: number, duration?: number): void {
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.el.currentTime = this.currentTime;
+    this.audio.el.addEventListener('ended', this.handleEnded);
+    this.bufferPromise = new Promise(resolve => {
+      this.bufferResolve = resolve;
+    });
+
+    const time = this.currentTime;
+
+    this.audio.el
+      .play()
+      .then(() => this.bufferPromise!.then())
+      .then(() => {
+        this.timestamp = performance.now();
+
+        // We need to compensate for the time it took to load the buffer
+        // otherwise the audio will be out of sync of the timer we use to
+        // render updates
+        if (this.audio?.el) {
+          // This must not be notifying of this adjustment otherwise it can cause sync issues and near infinite loops
+          this.setCurrentTime(time);
```

Comment:
and don't we need to really compensate this time? player can be playing for some time already or that's impossible?

Reply:
This is compensating for the time it takes for the buffer to resolve, it captures it before this promise chain.

https://github.com/heartexlabs/label-studio-frontend/pull/1360/files#diff-d26452e66010ef645ec1623dbb83eff35c0a81c26cce13a14f0dd3e2331366d9R34

Reply:
This code is entirely the same as previous.

Reply:
I was just wondering about possible edge cases, but yeah, you're right, that's not related anyway!